### PR TITLE
Disable authenticated session

### PR DESCRIPTION
**Pull request by pettersamuelsen**
### Changes
- Added an optional `useLegacyAuthentication` to the `authorize()` function
- `useLegacyAuthentication` will be `false` by default
- Fixes https://github.com/auth0/react-native-auth0/issues/226 when `ASWebAuthenticationSession` or `SFAuthenticationSession` is not needed 

### References

- https://github.com/auth0/react-native-auth0/issues/226

### Testing

* [x] This change adds unit test coverage
* [x] This change has been tested on the latest version of the platform/language

### CODE REVIEW

**Comment by cocojoe** [comment]
@pettersamuelsen Please take a look at this for naming https://github.com/auth0/Auth0.swift/blob/master/Auth0/WebAuth.swift#L185

Also should be documented and called out as the real use case here is for non SSO.

File: android/src/main/java/com/auth0/react/A0Auth0Module.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public String getName() {
     }
 
     @ReactMethod
-    public void showUrl(String url, boolean closeOnLoad, Callback callback) {
+    public void showUrl(String url, boolean closeOnLoad, boolean disableAuthenticatedSession Callback callback) {
```
Comment (lbalmaceda) [inline android/src/main/java/com/auth0/react/A0Auth0Module.java]: Isn't this missing a `,`?

File: webauth/agent.js [hunk 2]
```diff
@@ -17,16 +17,21 @@ export default class Agent {
...
+      NativeModules.A0Auth0.showUrl(
+        url,
+        closeOnLoad,
+        disableAuthenticatedSession,
+        (error, redirectURL) => {
+          Linking.removeEventListener('url', urlHandler);
+          if (error) {
+            reject(error);
+          } else if (redirectURL) {
+            resolve(redirectURL);
```
Comment (lbalmaceda) [inline webauth/agent.js]: what about joining the last 2 cases like this?

```
else if (redirectURL || closeOnLoad) {
  resolve(redirectURL);
}
```

File: webauth/index.js [hunk 3]
```diff
@@ -52,7 +52,7 @@ export default class WebAuth {
    *
    * @memberof WebAuth
    */
-  authorize(options = {}) {
+  authorize({ disableAuthenticatedSession, ...options } = {}) {
```
Comment (lbalmaceda) [inline webauth/index.js]: Why do you need to define it here? Assume is present inside the `options` object and that's it. This is only used on iOS and this method is for a generic webauth.

File: android/src/main/java/com/auth0/react/A0Auth0Module.java (hunk 1 above)
Comment (lbalmaceda) [inline android/src/main/java/com/auth0/react/A0Auth0Module.java]: comma missing between params.

File: webauth/agent.js (hunk 2 above)
Comment (pettersamuelsen) [reply webauth/agent.js]: This code is untouched in this PR, so I'll leave it outside of the scope of this PR

**Comment by pettersamuelsen** [comment]
@lbalmaceda  @cocojoe  I agree with your comments - see the updated commits

**Comment by cocojoe** [comment]
Have had a think about this PR and I don't think there is a real need to support this. The flow employed is correct and on the iOS side is part of Apple's guidance and is the path going forward.

**Pull request closed without merging**